packages/amplify_core/lib/src/types/auth/attribute/user_attribute_key.dart

```diff
@@ -21,10 +21,10 @@ import 'package:meta/meta.dart';
 @immutable
 abstract class UserAttributeKey implements Comparable<UserAttributeKey> {
   /// {@macro amplify_core.user_attribute_key}
-  const UserAttributeKey();
+  const UserAttributeKey(this.key);
 
   /// The JSON key for this attribute.
-  String get key;
+  final String key;
```

Comment:
I kinda prefer the other way so that you can define `key` as a field or a method in subclasses.